sympy/core/assumptions.py

```diff
@@ -196,7 +196,16 @@ class StdFactKB(FactKB):
     """
     rules = _assume_rules
 
-    def __init__(self, facts=None):
+    def __init__(self, facts=None, nonuser_facts=None):
+        # save a copy of the facts dict, then add the extra facts
+        #print((facts, type(facts), nonuser_facts))
+        if type(facts) is dict:  # not a subclass
```

Comment:
This is a bit horrid but gives the idea.  The `nonuser_facts` stuff is the "be strict about commutativity" from symbol.py; it is to prevent `commutative=True` from being added to all srepr output.